From dea04f5b5dc20bb62e9c01bb0355844ea0545a6e Mon Sep 17 00:00:00 2001
From: nxf77715 <shrivatsa.hande@nxp.com>
Date: Wed, 25 Sep 2024 12:06:40 +0530
Subject: [PATCH 15/18] [artf1182325]: Fix for NFC is not turning on after VTS
 Test

- Vts suite has test case PreDiscoverAfterClose which tests preDiscover
  api in NFC close state.
- Once preDiscover is called, SMB wired module initialize will be invoked
- Since NFC is off, SMB initialization command write will be failed. But
  credit window is decremented.
- Since we will not receive credit notification in this case subsequent
  HCI command will wait for credit window availability and blocks
  indefinitely causing watchdog to trigger
- To fix this HAL OPEN status is checked before calling SMB init

Change-Id: I08a2fb2dd4a92024d98adf1f30eae5c9c0ccb94e
---
 snxxx/halimpl/hal/phNxpNciHal.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/snxxx/halimpl/hal/phNxpNciHal.cc b/snxxx/halimpl/hal/phNxpNciHal.cc
index d4037e2d..486fc7ea 100644
--- a/snxxx/halimpl/hal/phNxpNciHal.cc
+++ b/snxxx/halimpl/hal/phNxpNciHal.cc
@@ -2325,8 +2325,10 @@ static void phNxpNciHal_core_initialized_complete(NFCSTATUS status) {
  *
  ******************************************************************************/
 int phNxpNciHal_pre_discover(void) {
-  phNxpNciHal_WiredSeDispatchEvent(&gWiredSeHandle, NFC_STATE_CHANGE,
-                                   (WiredSeEvtData)NfcState::NFC_ON);
+  if (nxpncihal_ctrl.halStatus != HAL_STATUS_CLOSE) {
+    phNxpNciHal_WiredSeDispatchEvent(&gWiredSeHandle, NFC_STATE_CHANGE,
+                                     (WiredSeEvtData)NfcState::NFC_ON);
+  }
   /* Nothing to do here for initial version */
   // This is set to return Failed as no vendor specific pre-discovery action is
   // needed in case of HalPrediscover
-- 
2.47.0

